src/language-markdown/printer-markdown.js

```diff
@@ -84,15 +84,18 @@ function genericPrint(path, options, print) {
         return "";
       }
       return [
-        normalizeDoc(printRoot(path, options, print)),
+        printRoot(path, options, print),
         !TRAILING_HARDLINE_NODES.has(getLastDescendantNode(node).type)
           ? hardline
           : "",
       ];
-    case "paragraph":
-      return printChildren(path, options, print, {
-        postprocessor: fill,
-      });
+    case "paragraph": {
+      const printed = cleanDoc(printChildren(path, options, print));
+      if (typeof printed === "string") {
+        return printed;
+      }
+      return fill(getDocParts(printed));
```

Comment:
Passing the result of `cleanDoc` to `fill` is wrong. `cleanDoc` doesn't guarantee that the array has the expected structure.

Reply:
Yes, you are  right, and it seems I've done something wrong in other places, I'll check.